File: wear/src/main/java/technology/mainthread/apps/gatekeeper/injector/module/AppModule.java
package technology.mainthread.apps.gatekeeper.injector.module;

import android.app.Application;
import android.content.Context;
import android.content.res.Resources;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private final Context context;

    public AppModule(Application application) {
        this.context = application.getApplicationContext();
    }

    @Provides
    @Singleton
    Context provideApplicationContext() {
        return context.getApplicationContext();
    }

    @Provides
    @Singleton
    Resources provideApplicationResources() {
        return context.getResources();
    }

}

File: wear/src/main/java/technology/mainthread/apps/gatekeeper/injector/module/AppModule.kt
package technology.mainthread.apps.gatekeeper.injector.module

import android.app.Application
import android.content.Context
import android.content.res.Resources
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(application: Application) {

    private val context: Context = application.applicationContext

    @Provides
    @Singleton
    internal fun provideApplicationContext(): Context {
        return context.applicationContext
    }

    @Provides
    @Singleton
    internal fun provideApplicationResources(): Resources {
        return context.resources
    }

}
